plugins/apollo-explorer/src/components/ApolloExplorerBrowser/ApolloExplorerBrowser.tsx

```diff
@@ -56,9 +58,42 @@ export type ApolloEndpointProps = {
 
 type Props = {
   endpoints: ApolloEndpointProps[];
+  authCallback?: () => Promise<string>;
+  apiHolder?: ApiHolder;
 };
 
-export const ApolloExplorerBrowser = ({ endpoints }: Props) => {
+export const handleAuthRequest = ({
+  legacyIncludeCookies,
+  authCallback,
+}: {
+  legacyIncludeCookies?: boolean;
+  authCallback: any;
+}): HandleRequest => {
+  let cookies = {};
+  if (legacyIncludeCookies) {
+    cookies = { credentials: 'include' };
+  } else if (legacyIncludeCookies !== undefined) {
+    cookies = { credentials: 'omit' };
+  } else {
+    cookies = {};
+  }
+
+  const handleRequestWithCookiePref: HandleRequest = async (
+    endpointUrl,
+    options,
+  ) =>
+    fetch(endpointUrl, {
+      ...options,
+      headers: {
+        ...options.headers,
+        Authorization: `Bearer ${await authCallback({})}`,
```

Comment:
This'll need an update for the return type tweak, but you can also skip passing `{}` here, mainly just confusing to the reader 😁

Reply:
[a1e39ff](https://github.com/backstage/backstage/pull/23405/commits/a1e39ffd2a711b6eb3129c497b2e2b11f059e429) [6ecacd3](https://github.com/backstage/backstage/pull/23405/commits/6ecacd3ffec0a25663803337363ad59d0d189906)